internal/querycoord/impl.go

```diff
@@ -228,18 +228,18 @@ func (qc *QueryCoord) LoadCollection(ctx context.Context, req *querypb.LoadColle
 		return status, nil
 	}
 
-	if collectionInfo, err := qc.meta.getCollectionInfoByID(collectionID); err == nil {
+	if collection, err := qc.meta.addCollection(collectionID, querypb.LoadType_LoadCollection, req.ReplicaNumber, req.Schema); err == ErrCollectionLoaded {
 		// if collection has been loaded by load collection request, return success
-		if collectionInfo.LoadType == querypb.LoadType_LoadCollection {
-			if collectionInfo.ReplicaNumber != req.ReplicaNumber {
+		if collection.LoadType == querypb.LoadType_LoadCollection {
+			if collection.ReplicaNumber != req.ReplicaNumber {
```

Comment:
Using collection value while error returned is not a recommended coding style.
Also, `errors.Is` is recommended to check error.

Reply:
1. It may be necessary here for atomicity. io.Reader returns a valid value even with error io.EOF, so it's acceptable for some special cases.
2. Fixed